pydexec/command.py

```diff
@@ -46,19 +46,16 @@ def env_to_arg(self, env_key, default=None, required=False, remove=True):
         :param required: raise an error if the environment variable is not set
         :param remove: remove the variable from the command's environment
         """
-        env_val = self._env_get(env_key, remove, default)
-        if env_val is not None:
-            self._args.append(env_val)
-        elif required:
+        if not self._arg_from_env(env_key, remove, default) and required:
```

Comment:
I don't really like having the real work here be a side effect of a conditional check. My first instinct on reading this was to rearrange the condition to `if required and not self._arg_from_env(...)` which is easier to read, but that would change the behaviour.

I also think the exception should be in the refactored bit. I'll try think of a clean way to do this, but it might be better just to drop the attempt to factor this out. :-/